setup.sh

```diff
@@ -1,106 +1,209 @@
+
+    [[ "${#missing_deps[@]}" -gt 0 ]] &&
+      return 1
+  else
+    for cmd in "${needed_docs_commands[@]}"; do
+      [[ $(command_exists "$cmd") ]] || missing_deps+=("$cmd")
+    done
+
+    while IFS='' read -r package; do
+      python3 -c "import pkg_resources; pkg_resources.require('$package')" &> /dev/null
+      [[ $? != 0 ]] && missing_pip_docs_deps+=("$package")
+    done < "documentation/dependencies/pip-docs.dependencies"
+
+    [[ "${#missing_deps[@]}" -gt 0 ||
+      "${#missing_pip_docs_deps[@]}" -gt 0 ]] &&
+      return 1
+  fi
+
+  return 0
+}
+
+function install_dependencies()
+{
+  deps_type="${1:-runtime}"
+
+  local package_list=()
   local cmd=''
+  local docs_str=''
+  local installed_cmd=''
   local distro
 
+  check_dependencies "$deps_type"
+
+  ret=$?
+
+  [[ $ret = 0 ]] && return 0
+
   distro=$(detect_distro '/')
+  supported_distros=(
+    'arch'
+    'debian'
+    'fedora'
+  )
+
+  if [[ ! ${supported_distros[*]} =~ $distro ]]; then
+    say 'User system not officially supported'
+    say 'Please install these packages:'
+    for pkg in "${!missing_deps[@]}"; do
+      printf '%s\n' "  - $pkg (provides ${missing_deps[$pkg]})"
+    done
+    if [[ "$deps_type" = 'docs' ]]; then
+      say 'And those pip packages:'
+      for pkg in "${!missing_pip_docs_deps[@]}"; do
+        printf '%s\n' "  - $pkg (provides ${missing_pip_docs_deps[$pkg]})"
+      done
+    fi
+    return 1
+  fi
+
+  [[ "$deps_type" != 'runtime' ]] &&
+    docs_str='-docs'
 
   if [[ "$distro" =~ 'arch' ]]; then
-    while IFS='' read -r package; do
-      installed=$(pacman -Qs "$package" > /dev/null)
-      [[ "$?" != 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/arch.dependencies"
-    cmd="pacman -S $package_list"
+    installed_cmd="pacman -Qs"
+    cmd="pacman -S ${package_list[*]}"
   elif [[ "$distro" =~ 'debian' ]]; then
-    while IFS='' read -r package; do
-      installed=$(dpkg-query -W --showformat='${Status}\n' "$package" 2> /dev/null | grep -c 'ok installed')
-      [[ "$installed" -eq 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/debian.dependencies"
-    cmd="apt install -y $package_list"
-  elif [[ "$distro" =~ 'fedora' ]]; then
-    while IFS='' read -r package; do
-      installed=$(dnf list installed "$package" 2> /dev/null | grep -c "$package")
-      [[ "$installed" -eq 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/fedora.dependencies"
-    cmd="dnf install -y $package_list"
+    installed_cmd="dpkg-query -W --showformat='${Status}\n"
+    cmd="apt install -y ${package_list[*]}"
   else
-    warning 'Unfortunately, we do not have official support for your distro (yet)'
-    warning 'Please, try to find the following packages:'
-    warning "$(cat "$DOCUMENTATION/dependencies/arch.dependencies")"
-    return 0
+    installed_cmd="dnf list installed"
+    cmd="dnf install -y ${package_list[*]}"
```

Comment:
Let's keep the fedora check, I know it's redundant but it adds clarity to the code and wouldn't really impact performance.